state/protocol/badger/validity.go

```diff
@@ -27,14 +27,16 @@ func isValidEpochSetup(setup *flow.EpochSetup) error {
 		identLookup[participant.NodeID] = struct{}{}
 	}
 
-	// there should be no duplicate node addresses
-	addrLookup := make(map[string]struct{})
-	for _, participant := range setup.Participants {
-		_, ok := addrLookup[participant.Address]
-		if ok {
-			return fmt.Errorf("duplicate node address (%x)", participant.Address)
+	if !skipNetworkAddressValidations {
+		// there should be no duplicate node addresses
+		addrLookup := make(map[string]struct{})
+		for _, participant := range setup.Participants {
+			_, ok := addrLookup[participant.Address]
+			if ok {
+				return fmt.Errorf("duplicate node address (%x)", participant.Address)
+			}
+			addrLookup[participant.Address] = struct{}{}
```

Comment:
@vishalchangrani still not sure why we can't just remove address validation entirely? Wouldn't we just need to remove this piece of code?

Reply:
this is currently used to safeguard against duplicate addresses in the root snapshot file. I wanted to keep those existing validations untouched.